assembly/src/assembler/instruction/procedures.rs

```diff
@@ -68,42 +69,62 @@ impl Assembler {
                             })
                         }
                     })?;
-                context.register_external_call(&proc, false, mast_forest)?;
+                context.register_external_call(&proc, false, mast_forest_builder.forest())?;
+            }
+            Some(proc) => {
+                context.register_external_call(&proc, false, mast_forest_builder.forest())?
             }
-            Some(proc) => context.register_external_call(&proc, false, mast_forest)?,
             None if matches!(kind, InvokeKind::SysCall) => {
                 return Err(AssemblyError::UnknownSysCallTarget {
                     span,
-                    source_file: current_source_file,
+                    source_file: current_source_file.clone(),
                     callee: mast_root,
                 });
             }
             None => context.register_phantom_call(Span::new(span, mast_root))?,
         }
 
         let mast_root_node_id = {
-            // Note that here we rely on the fact that we topologically sorted the procedures, such
-            // that when we assemble a procedure, all procedures that it calls will have been
-            // assembled, and hence be present in the `MastForest`. We currently assume that the
-            // `MastForest` contains all the procedures being called; "external procedures" only
-            // known by digest are not currently supported.
-            let callee_id = mast_forest
-                .get_node_id_by_digest(mast_root)
-                .unwrap_or_else(|| panic!("MAST root {} not in MAST forest", mast_root));
-
             match kind {
-                // For `exec`, we return the root of the procedure being exec'd, which has the
-                // effect of inlining it
-                InvokeKind::Exec => callee_id,
-                // For `call`, we just use the corresponding CALL block
+                InvokeKind::Exec => {
```

Comment:
I'm confused how we have ended up back here again - `exec` and `call` are _identical_ in every respect, with the sole exception of their execution context semantics. Why are we treating `exec` differently than `call` here?

Reply:
As discussed offline, this was updated to act similarly to call: [25fe82f](https://github.com/0xPolygonMiden/miden-vm/pull/1359/commits/25fe82ff8a7e8b80b89c87337f9352e11297f633)